src/views/documents/SearchProvider.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useContext, useState, useEffect } from "react";
+import { Outlet } from "react-router-dom";
+import { useSearchParams } from "react-router-dom";
+import useClient from "../../hooks/useClient";
+import { JournalsStoreContext } from "../../hooks/useJournalsLoader";
+import { SearchStore, SearchStoreContext } from "./SearchStore";
+import { LayoutDummy } from "../../layout";
+
+export function SearchProvider(props: any) {
+  const jstore = useContext(JournalsStoreContext);
+  const client = useClient();
+  const [params, setParams] = useSearchParams();
+  const [searchStore, setSearchStore] = useState<null | SearchStore>(null);
+
+  // This is more like an effect. This smells. Maybe just roll this all up into
+  // a hook.
```

Comment:
Probably should remove this (copied from container); this component is sufficiently special purpose.